packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4375,11 +4375,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   TextSelection? _adjustedSelectionWhenFocused() {
     TextSelection? selection;
-    final bool shouldSelectAll = widget.selectionEnabled && kIsWeb
-        && !_isMultiline && !_nextFocusChangeIsInternal;
+    final bool isDesktop = switch (defaultTargetPlatform) {
+      TargetPlatform.android || TargetPlatform.iOS || TargetPlatform.fuchsia => false,
+      TargetPlatform.macOS || TargetPlatform.linux || TargetPlatform.windows => true,
+    };
+    final bool shouldSelectAll = widget.selectionEnabled
+        && (kIsWeb || isDesktop)
+        && !_isMultiline
+        && !_nextFocusChangeIsInternal;
```

Comment:
Could you leave a comment explaining what platforms you tested here? For the future in case someone considers updating this code.

Reply:
There was an existing comment on line 4387 that I updated do you think this comment is enough? I checked that this is the expected behavior on Linux, macOS and Windows, and also that is not expected on iOS and Android.